keps/sig-network/20191205-kube-proxy-IP-node-binding.md

```diff
@@ -0,0 +1,70 @@
+---
+title: Give the ability to the LoadBalancer provisioner to disable the LB IP binding
+authors:
+  - "@Sh4d1"
+owning-sig: sig-network
+participating-sigs:
+  - sig-cloudprovider
+reviewers:
+  - "@thockin"
+  - "@cheftako"
+  - TBD
+approvers:
+  - "@thockin"
+  - TBD
+editor: TBD
+creation-date: 2019-12-05
+last-updated: 2019-12-05
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# Give the ability to the LoadBalancer provisioner to disable the LB IP binding
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+<!-- /toc -->
+
+## Summary
+
+The different kube proxy implentations (at least ipvs and iptables) are today sort of binding the External IPs of the LoadBalancer Service to each node. (iptables is creating some rules to redirect directly the packets to the service and ipvs is binding the IP to one interface on the node). This feature is exisiting because:
+
+1. To redirect pod traffic that is intended for LoadBalancer IP directly to the service, bypassing the loadbalancer for path optimization.
+2. Some LBs are sending packet with the destination IP equal to the LB IP, so the IP needs to be routable directly to the right service (otherwise looping back and forth)
```

Comment:
I'm not sure this is accurate. There are two different rules. One rule does a match for incoming traffic whose destination is Svc LB IP and jumps to the service chain. There is another rule that determines whether we need to masquerade the traffic or not before going to the service chain based on whether it originated from the cluster or outside.

AFAIK, sending a packet with destination address as the LB IP does not loop and works correctly.

Reply:
Yep with the current behaviour, sending a packet from the cluster to the LB is catched and redirected to the service IP. But if the behaviour changes to always go through the LB I think it might break GKE for instance (what I understood from @thockin ).

Reply:
not break, as much as be a different behavior from the current implementation, as far as I understand.

Reply:
Not so familiar with GKE :sweat_smile: I'll gladly take a sugestion for a replacement :)